docs: audit log provider (#2049)
pankod/refine: documentation/docs/core/hooks/audit-log/useLog.md

---
id: useLog
title: useLog
---

> If you need to create an audit log, **refine** provides the useLog hook for it. Returns two mutations `log` and `rename`, the method to create or update from [`auditLogProvider`](/core/providers/audit-log-provider.md#create) under the hood.

Reviewer: ```suggestion
If you need to create an audit log, **refine** provides the useLog hook for it. Returns two mutations `log` and `rename` that the methods uses `create` or `update` from [`auditLogProvider`](/core/providers/audit-log-provider.md#create) under the hood.
```

```tsx
import { useLog } from "@pankod/refine-core";

const { log, rename } = useLog();

```

> ## `log`
> 
> ```tsx

Reviewer: ```suggestion
## `log`

If you need to create an audit-log, you can use the `useLog` hook's `create` method. It returns the `create` method from [auditLogProvider](https://pankod-refine-documentation-pr-2049.surge.sh/docs/core/providers/audit-log-provider/) under the hood.

```tsx
```

| TError     | Custom error object that extends [`HttpError`](/core/interfaces.md#httperror)       | [`HttpError`](/core/interfaces.md#httperror)   | [`HttpError`](/core/interfaces.md#httperror)   |
| TVariables | Values for mutation function                                                        | `{}`                                           | `{}`                                           |

### Return value

| Description                               | Type                                                                                                                                                                      |
| ----------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Result of the `react-query`'s useMutation | [`UseMutationResult<`<br/>`{ data: TData},`<br/>`TError,`<br/>` { id: BaseKey; name: string; },`<br/>` unknown>`](https://react-query.tanstack.com/reference/useMutation) |

## Rename

> If you need to update the audit-log, you can use the `useLog` hook `rename` method. It returns the `update` method from [`auditLogProvider`](/core/providers/audit-log-provider.md) under the hood.

Reviewer: ```suggestion
If you need to update the audit-log, you can use the `useLog` hook's `rename` method. It returns the `update` method from [`auditLogProvider`](/core/providers/audit-log-provider.md) under the hood.
```

---
id: useLog
title: useLog
---

> If you need to create an audit log, **refine** provides the useLog hook for it. Returns two mutations `log` and `rename` that the methods uses `create` or `update` from [`auditLogProvider`](/core/providers/audit-log-provider.md#create) under the hood.

Reviewer 2: ```suggestion
If you need to create an audit log, **refine** provides the `useLog` hook for it. Returns two mutations `log` and `rename` that the methods uses `create` or `update` from [`auditLogProvider`](/core/providers/audit-log-provider.md#create) under the hood.
```

```tsx
import { useLog } from "@pankod/refine-core";

const { log, rename } = useLog();

```

## `log`

> If you need to create an audit-log, you can use the `useLog` hook's `create` method. It returns the `create` method from [auditLogProvider](https://pankod-refine-documentation-pr-2049.surge.sh/docs/core/providers/audit-log-provider/) under the hood.

Reviewer 3: ```suggestion
If you need to create an audit log, you can use the `useLog` hook's `create` method. It returns the `create` method from [auditLogProvider](https://pankod-refine-documentation-pr-2049.surge.sh/docs/core/providers/audit-log-provider/) under the hood.
```
